tests/unit/deploy/test_swagger.py

```diff
@@ -41,6 +41,96 @@ def test_can_produce_doc_for_method(sample_app, swagger_gen):
     # Same for x-amazon-apigateway-integration.
 
 
+def test_can_produce_doc_for_no_docstring(sample_app, swagger_gen):
+    @sample_app.route('/method')
+    def method():
+        pass
+
+    doc = swagger_gen.generate_swagger(sample_app)
+    view_config = doc['paths']['/method']['get']
+    assert 'summary' not in view_config
+    assert 'description' not in view_config
+
+
+def test_can_produce_doc_for_single_docstring(sample_app, swagger_gen):
+    @sample_app.route('/method1')
+    def method1():
+        """Single line method summary"""
+        pass
+
+    @sample_app.route('/method2')
+    def method2():
+        '''Single line method summary'''
+        pass
+
+    @sample_app.route('/method3')
+    def method3():
+        """
+        Single line method summary
+        """
+        pass
+
+    doc = swagger_gen.generate_swagger(sample_app)
+    for method in [1, 2, 3]:
+        view_config = doc['paths']['/method' + str(method)]['get']
+        assert 'summary' in view_config
+        assert 'description' not in view_config
+        assert view_config['summary'] == 'Single line method summary'
+
+
+def test_can_produce_doc_for_multi_docstring(sample_app, swagger_gen):
+    @sample_app.route('/method1')
+    def method1():
+        """Multiline method summary
+
+        And here is a more detailed description that can span multiple
+        lines. It can also handle indenting for things like method
+        arguments like so:
+            param1 - description
+            param2 - description
+        """
+        pass
+
+    @sample_app.route('/method2')
+    def method2():
+        """Multiline method summary
+
+            And here is a more detailed description that can span multiple
+            lines. It can also handle indenting for things like method
+            arguments like so:
+                param1 - description
+                param2 - description
+        """
+        pass
+
+    @sample_app.route('/method3')
+    def method3():
+        """Multiline method summary
+        And here is a more detailed description that can span multiple
+        lines. It can also handle indenting for things like method
+        arguments like so:
+            param1 - description
+            param2 - description
+
+
+        """
+        pass
+
+    doc = swagger_gen.generate_swagger(sample_app)
```

Comment:
Could we add a test that actually deploys some documentation and then uses the API to confirm that it is set correctly?

Reply:
Great idea, but I'm not sure how to do that in the framework of your tests. I tried to follow the model of tests that were already there. If you could point me to an example in the code that would help a lot.

Reply:
The integration tests `test_features.py` should be what you want. It deploys `test/integration/testapp/` and then uses various API methods to determine if it was deployed correctly.

 So you can add a new method to that app that has a documentation string and then write a test similar to 

```python
def test_path_params_mapped_in_api(smoke_test_app, apig_client):
```

Which gives you the deployed app and a client to interact with the APIG API.

Reply:
I'll check that out.

Reply:
Got it all in, and happy with the results. Sorry for the lint fails that I should have checked first.